pkg/controller/dual-pods/inference-server.go

```diff
@@ -578,14 +685,10 @@ func SliceMap[Domain, Range any](slice []Domain, mapFn func(Domain) (Range, erro
 	return mapped, errors
 }
 
-func IterFilter[Elt any](it iter.Seq[Elt], include func(Elt) bool) iter.Seq[Elt] {
-	return func(yield func(Elt) bool) {
-		for elt := range it {
-			if include(elt) {
-				if !yield(elt) {
-					return
-				}
-			}
-		}
+func SliceRemoveOnce[Elt comparable](slice []Elt, goner Elt) ([]Elt, bool) {
```

Comment:
Can you move this function (and `SliceMap`) to a different file as they are both generic utilities.